retail/doc-samples/search.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud import retail_v2
+
+def search_products(project_id, catalog_id, placement_id, branch, visitor_id, query):
+    client = retail_v2.SearchServiceClient()
+
+    placement_name = retail_v2.SearchServiceClient.placement_path(
+        project_id, location='global', catalog=catalog_id, placement=placement_id
+    )
+
+    search_request = retail_v2.SearchRequest(
+        placement=placement_name,
+        branch=branch,
+        visitor_id=visitor_id,
+        query=query,
+        page_size=10
+    )
+
+    response = client.search(search_request)
+
+    print(f"Search response: {response}")
+
+    return response
+
+# Example usage
+project_id = 'YOUR_PROJECT_ID'
+catalog_id = 'YOUR_CATALOG_ID'
+placement_id = 'YOUR_PLACEMENT_ID'
+branch = 'YOUR_BRANCH_NAME'
+visitor_id = 'YOUR_VISITOR_ID'
+query = 'YOUR_QUERY'
+
+search_products(project_id, catalog_id, placement_id, branch, visitor_id, query)
```

Comment:
This sample does not match our styleguide, particularly with respect to testing: https://googlecloudplatform.github.io/samples-style-guide/#testing

You will need to also add a region tag around the part of the code you want to include in the documentation. 

I would also suggest you move this sample to a different folder, as `doc-samples` isn't very clear. I would suggest `retail/snippets/search_request.py`, and you can then create `retail/snippets/search_request_test.py` for the test. You can also use this folder for future samples with a shared `requirements.txt` another configs (which you have also missed)